### ISSUE TITLE
`flutter create package` should generate UI code for each platform

### ISSUE THREAD

**Issue by mgenware** [flutter/flutter#139298]
### Is there an existing issue for this?

- [X] I have searched the [existing issues](https://github.com/flutter/flutter/issues)
- [X] I have read the [guide to filing a bug](https://flutter.dev/docs/resources/bug-reports)

### Use case

After #119844 is resolved. There seems to be no easy way to create a flutter **widget** package. Now `flutter create package` creates a dart package with no example folder to run a demo app on each platform, and no widget test files. While the `pubspec.yaml` in generated project still depends on flutter though.

### Proposal

Developers should be encouraged to create dart packages via `dart create package`. `flutter create package` should create a full widget package with widget test files, and an example project to quickly test the widget on each platform.

**Comment by christopherfujino** [comment]
Can you clarify: "should create example projects for each platform"? Do you mean like, separate example apps for android, ios, web, etc?

**Comment by mgenware** [comment]
@christopherfujino yes! flutter used to generate example projects for `flutter create --template=package`, now it's gone.

Tried the latest flutter 3.24.0, there's test files now. but still no way to run the project without adding an example project manually.

![image](https://github.com/user-attachments/assets/c0bbc053-8c7b-424c-9f95-f32c3ae0d91b)

**Comment by bkonyi** [comment]
We're unlikely to prioritize this work ourselves, but if you'd like to upload a PR adding an example project to the `package` templates, I'd be happy to review it!
